stacker_blueprints/vpc.py

```diff
@@ -248,21 +248,25 @@ def create_network(self):
                     # Private subnets are where actual instances will live
                     # so their gateway needs to be through the nat instances
                     if variables["UseNatGateway"]:
-                        instance_id = NOVALUE
                         nat_gateway_id = Ref(NAT_GATEWAY_NAME % name_suffix)
+                        t.add_resource(
```

Comment:
I think we can support nat instances and gateways without having to duplicate the `ec2.Route` - please see me other comment. 

Lets try to do something like this:

```
                        route= ec2.Route(
                            route_name,
                            RouteTableId=Ref(route_table_name),
                            DestinationCidrBlock='0.0.0.0/0',
                        )
                        if variables["UseNatGateway"]:
                            route.NatGatewayId = Ref(NAT_GATEWAY_NAME % name_suffix)
                        else:
                            route.InstanceId = Ref(NAT_INSTANCE_NAME % name_suffix)
                        t.add_resource(route)
```